src/components/Section/Section.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import Box, { BoxProps } from '../Box';
+import { HeadingLevelContext } from '../Heading';
+
+/**
+ * Increases the heading level of all `Heading` components inside it by 1.
+ */
+export const Section: React.FC<BoxProps<'section'>> = ({ children, as = 'section', ...rest }) => {
+  const level = React.useContext(HeadingLevelContext);
```

Comment:
There it's! 
Also, I don't get why we have to increase the containing headings by 1? 
Sections should always have a heading, with very few exceptions. 
So an `h1` -> `h6` will do the job, but why don't we use an h2 over h3? 

I think it would be better to default the `hTag` to `h2` and don't use the context at all!